model/src/order.rs

```diff
@@ -33,6 +33,7 @@ pub struct Order {
     pub order_meta_data: OrderMetaData,
     #[serde(flatten)]
     pub order_creation: OrderCreation,
+    pub status: OrderStatus,
```

Comment:
should this be part of the `OrderMetaData` struct?

Reply:
There is a notable distinction that this is a "computed" field as opposed to one stored directly in the db. One thing I don't like about putting it in the metadata is that it is extremely easy to create invalid `OrderMetadata` instances with the current API:
```rust
OrderMetaData {
    invalidated: true,
    status: OrderStatus::Open,
}
```

This problem already exists with the current model (you can have `OrderMetaData` that is invalid for its `OrderCreation`).

Just wanted to bring this up, I don't think this is something that we should necessarily solve in this PR.

Reply:
So, I have the same concern as Nick.
I rather not have it in the ordermetada nor ordercreation structs as this is a calculated field.
I would rather have a third struct like `DerivedFields` or whatever to distinguish between what goes in the db and what doesn't.

What are your thoughts on this?

Reply:
In the http api this does not matter because all the fields get flattened anyway. So this would be purely from the rust perspective. Originally the idea behind metadata was to contain everything that isn't order creation. Metadata already has fields that are not stored directly as values in the db like the balance.
Either way you can still create invalid combined orders so I don't feel moving the fields changes much in that regard. If we wanted to prevent that then it shouldn't be a field and instead be calculated by users of the api.
Anyway, I don't have a problem with a new DerivedFields struct either so if it's fine to do it that way if more like prefer it.

Reply:
Invalidate is only needed for backwards compatibility, isn't it? Technically it's superseded by OrderStatus. In a way, because order status can be derived from the fields on the order already we create this data redundancy which also allows for inconsistent state (I assume the reason we do it here is that we don't have to do it on the client multiple times)?

Generally I'm not sure we should derive many things in the backend (and maybe instead offer a core library on the ts side that can derive things from the original json) so I would also vote for keeping this part of the metadata.

Reply:
Makes sense. I agree and think there's room for improvement separating the fields by "concern", but then it wouldn't fit in this PR.
For now, I'll move it to OrderMetaData.